TypeSystem/flow-analysis/reachability_A05_t01.dart

```diff
@@ -2,26 +2,21 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// @assertion other literal: If N is some other literal than the above, then:
-///  Let null(N) = unreachable(before(N)).
-///  Let notNull(N) = before(N)
+/// @assertion other literal: If `N` is some other literal than the above, then:
+/// - Let `after(N) = before(N)`.
 ///
-/// @description Checks reachability after string literal
+/// @description Checks that a string literal is always non-nullable.
 /// @author sgrekhov@unipro.ru
 /// @issue 41985
-
-import "../../Utils/expect.dart";
+/// @issue 60114
 
 main() {
   late int i;
-  if ("Lily was here" == null) {
-    i = 42;   // `i` is not definitely unassigned because in a weak mode the
-              // condition may be true
-  }
-  try {
-    i;
-    Expect.fail("Error expected");
-  } on Error catch(e) {
-    // Ok
+  if ("Lily was here" == null) { // ignore: unnecessary_null_comparison
+    i = 42;
   }
+  i; // Definitely unassigned
```

Comment:
OK, so this test assumes that we have already introduced the changes to the flow analysis that are discussed in dart-lang/language#3100.

@stereotype441, how do we coordinate this with a spec update? The [flow-analysis feature spec](https://github.com/dart-lang/language/blob/main/resources/type-system/flow-analysis.md) already refers to issue 3100 in one location, but there might be additional adjustments coming up as this is implemented, right?